File: db/person_language.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in person_language.
*/

class Person_Language extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "person_language";
    $this->order = "";
    $this->domain = "person";
    $this->field['person_id']['type'] = 'INTEGER';
    $this->field['person_id']['not_null'] = true;
    $this->field['language_id']['type'] = 'INTEGER';
    $this->field['language_id']['not_null'] = true;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['person_id']['primary_key'] = true;
    $this->field['language_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/person_language.rb
module Momomoto
  class Person_language < Base
    def initialize
      super
      @domain = 'person'
      @fields = {
        :person_id => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} ),
        :language_id => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
